hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/clientrm/TestFederationClientInterceptor.java

```diff
@@ -181,6 +195,15 @@ public void setUp() {
 
     interceptor.setConf(this.getConf());
     interceptor.init(user);
+    RouterDelegationTokenSecretManager tokenSecretManager =
+        interceptor.createRouterRMDelegationTokenSecretManager(this.getConf());
+    try {
+      tokenSecretManager.startThreads();
+      interceptor.setTokenSecretManager(tokenSecretManager);
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      Assert.fail();
```

Comment:
Why not just let the exception surface?

Reply:
Thanks for your suggestion, I will modify the code.